misc: Replacing LayerCI references with webapp.io (finishing previous PR)

joshuadsouza opened the pull request:
- Closes

### Additional details
LayerCI has rebranded to webapp.io, context here https://webapp.io/blog/layerci-has-rebranded-to-webapp-io/. 

The environment build variables have updated in webapp.io to some new variables (instead of the old ones with reference to LayerCI). Refer to the following documentation in webapp.io with the [`environment build variables`](https://docs.webapp.io/layerfile-reference/build-env)

This PR finishes up the work done in this previously opened PR https://github.com/cypress-io/cypress/pull/28577/files and also adds in the `EXPOSE_WEBSITE_HOST` build environment variable exposed by webapp.io 

### Steps to test

### How has the user experience changed?

### PR Tasks
- [x] Have tests been added/updated?
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/5632
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

joshuadsouza committed:
Replacing LayerCI references with webapp.io

joshuadsouza committed:
Update CI params exposed from webapp.io and add EXPOSE_WEBSITE_HOST

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cypress-io/cypress?pullRequest=28874) <br/>All committers have signed the CLA.

joshuadsouza committed:
Adding misc to changelog references changes from LayerCI to webappio

jennifer-shehane reviewed (changes requested):
@joshuadsouza What's the intention behind adding the `EXPOSE_WEBSITE_HOST`? I don't think this env var is needed. These env vars are primarily surfaced for Cypress use, and we don't have a need for this env var that I can see. 

Otherwise this looks good!

joshuadsouza committed:
Removing EXPOSE_WEBSITE_HOST

joshuadsouza commented:
> @joshuadsouza What's the intention behind adding the `EXPOSE_WEBSITE_HOST`? I don't think this env var is needed. These env vars are primarily surfaced for Cypress use, and we don't have a need for this env var that I can see.
> 
> Otherwise this looks good!

Got it, just updated the PR and removed `EXPOSE_WEBSITE_HOST` 👍

jennifer-shehane committed:
Merge branch 'develop' into develop

jennifer-shehane committed:
Merge branch 'develop' into develop
